### DISCUSSION TITLE
How to enable  CORS

### DISCUSSION THREAD

**Discussion by Petri-Leinonen** [Azure/data-api-builder#1627] [Q&A] [answered]
How can I enable the CORS when using Data API Builder?

I know that when I make a Static Web App, I can add the globalHeaders settings in staticwebapp.config.json -file, like this:

```
{
    "globalHeaders": {
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET, POST, PUT, OPTIONS"
    }
}
```

But how can I do the similar settings in Data API Builder? I'm getting this error because the CORS is not enabled:

`Access to fetch at 'https://myapp.net/data-api/api/devices/Id/c131eba55af6e8d8' from origin 'https://localhost' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

**Comment by severussundar** [comment, marked as answer]
CORS can be enabled by declaring a `cors` field within the `host` element in the config file.

The `--cors-origin` option of the `init` command can be used to setup the allowed origins. [Init command options](https://learn.microsoft.com/azure/data-api-builder/data-api-builder-cli#init)

Here's the link to the docs that describes the `host` section of the config file: [Host element docs](https://learn.microsoft.com/azure/data-api-builder/configuration-file#host)
